### ISSUE TITLE
when using the ipopt for nlp

### ISSUE THREAD

**Issue by studentlemon** [casadi/casadi#2521]
WARNING(".../casadi/core/plugin_interface.hpp:322: Assertion "handle!=nullptr" failed:
PluginInterface::load_plugin: Cannot load shared library 'libcasadi_nlpsol_ipopt.so': 
   (
    Searched directories: 1. casadipath from GlobalOptions
                          2. CASADIPATH env var
                          3. PATH env var (Windows)
                          4. LD_LIBRARY_PATH env var (Linux)
                          5. DYLD_LIBRARY_PATH env var (osx)
    A library may be 'not found' even if the file exists:
          * library is not compatible (different compiler/bitness)
          * the dependencies are not found
   )
  Tried '' :
    Error code: libcasadi_nlpsol_ipopt.so: cannot open shared object file: No such file or directory
  Tried '.' :
    Error code: ./libcasadi_nlpsol_ipopt.so: cannot open shared object file: No such file or directory") [.../casadi/core/plugin_interface.hpp:171]
terminate called after throwing an instance of 'casadi::CasadiException'
  what():  .../casadi/core/plugin_interface.hpp:417: Plugin 'ipopt' is not found.
Aborted (core dumped)

how to solver it?

**Comment by studentlemon** [comment]
> What's the mode of install? Can you find the library on your filesystem?

yeah,I have solved the above problems, but when i compile the project:
casadi/interfaces/ipopt/ipopt_nlp.hpp:46:64: fatal error: casadi/interfaces/ipopt/casadi_nlpsol_ipopt_export.h: No such file or directory
How to solver it?

**Comment by studentlemon** [comment]
> > What's the mode of install? Can you find the library on your filesystem?
> 
> yeah,I have solved the above problems, but when i compile the project:
> casadi/interfaces/ipopt/ipopt_nlp.hpp:46:64: fatal error: casadi/interfaces/ipopt/casadi_nlpsol_ipopt_export.h: No such file or directory
> How to solver it?
It's not in the folder

**Comment by studentlemon** [comment]
> What's the mode of install? Can you find the library on your filesystem?

install by src coding. yeah, but when i run the project, the error will be occur

**Comment by studentlemon** [comment]
> > What's the mode of install? Can you find the library on your filesystem?
> 
> install by src coding. yeah, but when i run the project, the error will be occur

Could not find a package configuration file provided by "Ipopt" with any of
  the following names:

    IpoptConfig.cmake
    ipopt-config.cmake

  Add the installation prefix of "Ipopt" to CMAKE_PREFIX_PATH or set
  "Ipopt_DIR" to a directory containing one of the above files.  If "Ipopt"
  provides a separate development package or SDK, be sure it has been
  installed.
 i have installed the Ipopt on my filesystem. but can not find the .cmake in the filesystem of Ipopt

**Comment by chjXu** [comment]
Have you solved this problem?　I have the same problem.

**Comment by GPrathap** [comment]
If you are using lubuntu or ubuntu after following instructions given here: https://github.com/casadi/casadi/wiki/InstallationLinux

`cmake -DWITH_IPOPT=true ..`  when you build it. And check whether it is found or not.

**Comment by trunc8** [comment]
I was installing the [path-optimizer package](https://github.com/LiJiangnanBit/path_optimizer) when I ran into the exact error mentioned by @studentlemon. Please excuse me if this solution isn't relevant to the Casadi package, but the modification suggested by this [link](https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/issues/266) solved my issue!

**Comment by ChenZhenY** [comment]
> > > What's the mode of install? Can you find the library on your filesystem?
> > 
> > 
> > install by src coding. yeah, but when i run the project, the error will be occur
> 
> Could not find a package configuration file provided by "Ipopt" with any of the following names:
> 
> ```
> IpoptConfig.cmake
> ipopt-config.cmake
> ```
> 
> Add the installation prefix of "Ipopt" to CMAKE_PREFIX_PATH or set "Ipopt_DIR" to a directory containing one of the above files. If "Ipopt" provides a separate development package or SDK, be sure it has been installed. i have installed the Ipopt on my filesystem. but can not find the .cmake in the filesystem of Ipopt

I have the same issue. And what confuses me is that, if you download the 3.12+ code from ipopt github release and compile, you won't have this .cmake files because they use only makefiles to build and install. However, if you install via sudo apt-get, the latest version you can get is 3.11.9. Anyone knows how to install 3.12+ ipopt so that we can have these .cmake files? Thanks in advance!
